instrumentation/github.com/bradfitz/gomemcache/example/client.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/bradfitz/gomemcache/memcache"
+
+	otelgomemcache "go.opentelemetry.io/contrib/instrumentation/github.com/bradfitz/gomemcache"
+	otelglobal "go.opentelemetry.io/otel/api/global"
+
+	oteltracestdout "go.opentelemetry.io/otel/exporters/stdout"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+func main() {
+	var host, port = os.Getenv("HOST"), "11211"
+
+	initTracer()
+	ctx := context.Background()
+
+	c := otelgomemcache.NewClientWithTracing(
+		memcache.New(
+			host + ":" + port,
+		),
+	)
+
+	t := otelglobal.Tracer("memcached-test")
+	ctx, s := t.Start(ctx, "test-operations")
+	doMemcacheOperations(ctx, c)
+	s.End()
+}
+
+func doMemcacheOperations(ctx context.Context, c *otelgomemcache.Client) {
+	err := c.WithContext(ctx).Add(&memcache.Item{
```

Comment:
Should this start with creating a single local copy of the `Client` with a `Context`?  I worry that using a pattern where each call requires adding the `WithContext(ctx)` may result in users attempting to retrofit existing memcache uses missing some operations.

```go
c := c.WithContext(ctx)
err := c.Add(...)
```

The primary downside I see with that approach is the potential for starting a span and forgetting to update the context for the `Client`.  Thoughts?

Reply:
I agree, this looks like a misleading example.

Reply:
I see now this example was not the best choice, adapted according to your recommendation, thank you :+1: